.github/workflows/infrastructure.yml

```diff
@@ -0,0 +1,173 @@
+
+name: Infrastructure 🔨
+run-name: APIM base infrastructure build from ${{ github.repository }}
+
+on:
+  push:
+    branches: [ APIM-149-ia-c-dev-environment ]
+    paths: [ .github/workflows/infrastructure.yml ]
+
+env:
+  product: apim
+  environment: infrastructure
+  timezone: 'Europe/London'
+  # Deployment environment target i.e., `dev`, `staging`, `prod` 
+  target: ${{ vars.environment }}
+
+jobs:
+
+# 1. Setup infrastructure variables
+  setup:
+    name: Setup 🔧
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    outputs:
+      environment: ${{ env.environment }}
+      timezone: ${{ env.timezone }}
+    steps:
+    - name: Environment 🧪
+      run: echo "Environment set to ${{ env.environment }}"
+
+    - name: Timezone 🌐
+      run: echo "Timezone set to ${{ env.timezone }}"
+
+# 2. Base infrastructure creation
+  base:
+    name: Base 🧱
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [ self-hosted, APIM, infrastructure ]
+    steps:
+
+    - name: Pre-production 💫
+      if: contains('["dev", "staging"]', env.target)
+      run: echo "type=Preproduction" >> $GITHUB_ENV
+
+    - name: Production 💫
+      if: ${{ 'prod' == env.target }}
+      run: echo "type=Production" >> $GITHUB_ENV
+
+    - name: Tags 🏷️
+      run: echo tags='Environment=${{ env.type }}' \
+        'Product=${{ env.product }}' \
+        'Region=${{ vars.REGION }}' \
+        'Team=development' >> $GITHUB_ENV
+
+    - name: Login 🔐
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Azure defaults ✨
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          # Basic
+          az configure --defaults location=${{ vars.REGION }}
+          az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
+
+          # AZ CLI extensions upgrade
+          az extension add --name containerapp --upgrade
+
+    - name: Resource group 🏗️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az group create \
+          --name rg-${{ env.product }}-${{ env.target }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container registry 📦️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az acr create \
+          --name cr${{ env.product }}${{ env.target }}${{ vars.REGION }}001 \
+          --sku Standard \
+          --admin-enabled true \
+          --tags ${{ env.tags }}
+
+    - name: Container app environment 🗃️
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az containerapp env create \
+          --name cae-${{ env.product }}-${{ env.target }}-${{ vars.REGION }}-001 \
+          --tags ${{ env.tags }}
+
+    
+    - name: Container app - MDM 📄
+      uses: azure/cli@v1.0.0
+      with:
+        inlineScript: |
+          az containerapp create \
+          --name ca-mdm-${{ env.product }}-${{ env.target }}-${{ vars.REGION }}-001 \
```

Comment:
I think the name would be: `rg-mdm-001`
We are not adding the region in the name

Reply:
It is a recommended naming convention by Azure and Regions are stored in GitHub so they are not visible. I can convert them to secrets so they do not appear in GHA execution output

Reply:
It's not that. We had a discussion internally  and agreed that the region should not be added to the name - this as part of DevSecOps